File: src/main/java/leetCode/offer/Offer53_2.java
package leetCode.offer;

/**
 * @author jingxinwu
 * @date 2021-11-24 1:15 上午
 */
public class Offer53_2 {

    public static void main(String[] args) {
        System.out.println(new Offer53_2().missingNumber(new int[]{7, 8, 9, 11, 12}));
        System.out.println(new Offer53_2().missingNumber(new int[]{1})); //0
        System.out.println(new Offer53_2().missingNumber(new int[]{1, 2})); //0
    }


    public int missingNumber(int[] nums) {
        int i = 0, j = nums.length - 1;
        while (i <= j) {
            int m = (i + j) / 2;
            if (nums[m] == m) {
                i = m + 1;
            } else {
                j = m - 1;
            }
        }
        return i;
    }
}

File: src/main/go/leetcode/lists/offer/Offer53_2.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2023/1/3 23:45
 */

func missingNumber(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	nums = append(nums, 0)
	for i := 0; i < len(nums)-1; i++ {
		index := Abs(nums[i])
		nums[index] = -nums[index]
	}
	for i := 0; i < len(nums); i++ {
		if nums[i] > 0 {
			return i
		}
	}
	return len(nums) - 1
}
